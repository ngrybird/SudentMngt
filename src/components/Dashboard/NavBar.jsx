import React from "react";
import profileAvatar from "../../images/avatar-header.png";
import vector from "../../images/Vector.png";

class NavBar extends React.Component {
  render() {
    return (
      <div className="navbar">
        <div className="profile">
          <span className="profile-info">
            <div className="profile-name" data-testid="id-profile-name">{this.props.name}</div>
            <div className="profile-designation">{this.props.position}</div>
          </span>
          <span className="profile-avatar">
            <img src={profileAvatar} height="50px" width="50px" alt="avatar" />
          </span>
          <span className="profile-more">
            <img src={vector} height="6px" width="12px" alt="avatar" />
          </span>
        </div>
      </div>
    );
  }
}
export default NavBar;
