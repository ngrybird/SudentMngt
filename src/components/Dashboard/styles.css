.dashboard-container {
  background: #f8f8fb;
  display: flex;
  flex-flow: column;
  flex-grow: 1;
}
.dashboard-header {
  margin-top: 66px;
}
.dashboard-text {
  margin-left: 40px;
  font-size: 30px;
  font-weight: 600;
}
.dashboard-subtext {
  margin-left: 40px;
  font-size: 14px;
  color: grey;
}
.navbar {
  padding-top: 27px;
  display: flex;
  flex-flow: column wrap;
  align-items: flex-end;
}
.profile {
  display: flex;
  width: 225px;
}
.profile-info {
  position: relative;
  right: 10px;
  top: 10px;
  text-align: right;
}
.profile-name {
  font-size: 14px;
}

.profile-designation {
  font-size: 12px;
}
.profile-more {
  position: relative;
  top: 10px;
  left: 10px;
}
.all-sections {
  display: flex;
  flex-flow: row wrap;
}
