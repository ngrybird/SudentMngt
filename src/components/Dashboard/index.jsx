import React from "react";
import NavBar from "./NavBar";
import Section from "../Section";
import StudentsTable from "../StudentsTable";
import "./styles.css";

class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    if (this.props.sectionsData) {
    }
  }
  render() {
    return (
      <div className="dashboard-container">
        <NavBar {...this.props.userProfile} />
        <div className="dashboard-header">
          <div className="dashboard-text">Dashboard</div>
          <div className="dashboard-subtext">Mobile UX/UI Design Course</div>
        </div>
        <div className="all-sections">
          {this.props.sectionsData &&
            this.props.sectionsData.map(section => <Section {...section} />)}
        </div>
        {this.props.studentsMarks && (
          <StudentsTable records={this.props.studentsMarks} />
        )}
      </div>
    );
  }
}
export default Dashboard;
