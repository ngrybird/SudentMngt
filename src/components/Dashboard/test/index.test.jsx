import React from "react";
import { render } from "@testing-library/react";
import Dashboard from "../index";
import { sectionsData, studentsMarks } from "./testData";

describe("<Dashboard/>", () => {
  const profile = {
    name: "Cody Simmons",
    position: "Lecturer"
  };
  it("check static elements on dashboard", () => {
    const { container, getByText, getAllByTestId } = render(
      <Dashboard
        userProfile={profile}
        sectionsData={sectionsData}
        studentsMarks={studentsMarks}
      />
    );
    const dashboardText = getByText("Dashboard");
    const USTitle = getByText("Mobile UX/UI Design Course");
    const sections = getAllByTestId("custom-sections");
    const profileName = getAllByTestId("id-profile-name");

    expect(dashboardText).toBeInTheDocument();
    expect(USTitle).toBeInTheDocument();
    expect(sections.length).toEqual(3);
    expect(profileName[0].innerHTML).toEqual("Cody Simmons");

    expect(container).toMatchSnapshot();
  });
});
