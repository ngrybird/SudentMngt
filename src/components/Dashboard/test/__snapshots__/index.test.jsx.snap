// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> check static elements on dashboard 1`] = `
<div>
  <div
    class="dashboard-container"
  >
    <div
      class="navbar"
    >
      <div
        class="profile"
      >
        <span
          class="profile-info"
        >
          <div
            class="profile-name"
            data-testid="id-profile-name"
          >
            Cody Simmons
          </div>
          <div
            class="profile-designation"
          >
            Lecturer
          </div>
        </span>
        <span
          class="profile-avatar"
        >
          <img
            alt="avatar"
            height="50px"
            src="avatar-header.png"
            width="50px"
          />
        </span>
        <span
          class="profile-more"
        >
          <img
            alt="avatar"
            height="6px"
            src="Vector.png"
            width="12px"
          />
        </span>
      </div>
    </div>
    <div
      class="dashboard-header"
    >
      <div
        class="dashboard-text"
      >
        Dashboard
      </div>
      <div
        class="dashboard-subtext"
      >
        Mobile UX/UI Design Course
      </div>
    </div>
    <div
      class="all-sections"
    >
      <div
        class="section-container"
        data-testid="custom-sections"
      >
        <div
          class="section-data"
        >
          <span>
            <img
              src="Ellipse.png"
            />
            <img
              class="hover-image"
              src="user.png"
            />
          </span>
          <div
            class="section-title"
          >
            62
          </div>
          <div
            class="section-subtitle"
          >
            Students
          </div>
        </div>
      </div>
      <div
        class="section-container"
        data-testid="custom-sections"
      >
        <div
          class="section-data"
        >
          <span>
            <img
              src="Ellipse.png"
            />
            <img
              class="hover-image"
              src="thunder.png"
            />
          </span>
          <div
            class="section-title"
          >
            6.8
          </div>
          <div
            class="section-subtitle"
          >
            Average Markes
          </div>
        </div>
      </div>
      <div
        class="section-container"
        data-testid="custom-sections"
      >
        <div
          class="section-data"
        >
          <span>
            <img
              src="Ellipse.png"
            />
            <img
              class="hover-image"
              src="thunder.png"
            />
          </span>
          <div
            class="section-title"
          >
            9
            <span
              class="section-subtext"
            >
              (14%)
            </span>
          </div>
          <div
            class="section-subtitle"
          >
            Underperforming students
          </div>
        </div>
      </div>
    </div>
    <div
      class="table-container"
    >
      <div
        class="table-header"
      >
        <span>
          Students by average marks
        </span>
        <button>
          Descending
        </button>
      </div>
      <hr />
      <div
        class="student-list"
      >
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            Annete Watson
          </span>
          <span
            class="row-marks"
          >
            9.3
          </span>
        </div>
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            Calvin Steward
          </span>
          <span
            class="row-marks"
          >
            8.9
          </span>
        </div>
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            Ralph Richards
          </span>
          <span
            class="row-marks"
          >
            8.7
          </span>
        </div>
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            Bernard Murphy
          </span>
          <span
            class="row-marks"
          >
            8.2
          </span>
        </div>
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            Arlene Robertson
          </span>
          <span
            class="row-marks"
          >
            8.2
          </span>
        </div>
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            John Lane
          </span>
          <span
            class="row-marks"
          >
            8.1
          </span>
        </div>
        <div
          class="table-row"
          data-testid="student-row"
        >
          <span>
            <img
              height="40px"
              src="Ellipse.png"
              width="40px"
            />
          </span>
          <span
            class="row-title"
          >
            Pat McKinney
          </span>
          <span
            class="row-marks"
          >
            7.9
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
